src/store/keystore.rs

```diff
@@ -142,11 +142,11 @@ impl<S: Store> Keystore for ClientKeystore<S> {
 
         let locations = [Location::Internal, Location::External, Location::Volatile];
 
-        secrecies.iter().any(|secrecy| {
-            let path = self.key_path(*secrecy, id);
-            locations
-                .iter()
-                .any(|location| store::delete(self.store, *location, &path))
+        locations.iter().rev().any(|location| {
```

Comment:
nit: changing the order in `locations` and `secrecies` would be more intuitive than reversing it